test/Integration/ApplicationTrait.php

```diff
@@ -27,9 +30,14 @@ trait ApplicationTrait
      */
     protected $failOnErrorEvents = true;
 
+    /**
+     * @param array<string, mixed> $extraConfig
+     */
     protected function setUpApplication(array $extraConfig = []): Application
     {
-        $extraConfig['service_manager']['services'][SendResponseListener::class] = new NoopSendResponseListener();
+        $listenerConfig = ['service_manager' => ['services' => [SendResponseListener::class => new Noop()]]];
+        $extraConfig    = array_merge($extraConfig, $listenerConfig);
```

Comment:
This change is breaking. It will replace `service_manager` content entirely while it is only meant to override `SendResponseListener` and nothing else. This have to be a recursive merge with stdlib array utils

I am not sure I agree with psalm complains on this one tbh. May be suppress?